[Zen2] Update documentation for Zen2 (#34714)
elastic/elasticsearch: docs/reference/modules/discovery/bootstrapping.asciidoc

[[modules-discovery-bootstrap-cluster]]
=== Bootstrapping a cluster

Starting an Elasticsearch cluster for the very first time requires the initial

> set of master-eligible nodes to be explicitly set on one or more of the

Reviewer: ```suggestion
set of <<master-node,master-eligible nodes>> to be explicitly set on one or more of the
```

master-eligible nodes in the cluster. This is known as _cluster bootstrapping_.
This is only required the very first time the cluster starts up: nodes that

> have already joined a cluster will store this information in their data folder,

Reviewer: ```suggestion
have already joined a cluster store this information in their data folder
```

> and freshly-started nodes that are intended to join an existing cluster will

Reviewer: ```suggestion
and freshly-started nodes that are joining an existing cluster
```

obtain this information from the cluster's elected master. This information is
given using this setting:

`cluster.initial_master_nodes`::

    Sets a list of the <<node.name,node names>> or transport addresses of the
    initial set of master-eligible nodes in a brand-new cluster. By default
    this list is empty, meaning that this node expects to join a cluster that
    has already been bootstrapped.

This setting can be given on the command line when starting up each

> master-eligible node, or added to the `elasticsearch.yml` configuration file on

Reviewer: This paragraph makes it sound like you'd have to do this on every master-eligible node, whereas the initial paragraph made it sound like you only had to do it on one of them.

--------------------------------------------------

Like all node settings, it is also possible to specify the initial set of
master nodes on the command-line that is used to start Elasticsearch:

[source,bash]
--------------------------------------------------
$ bin/elasticsearch -Ecluster.initial_master_nodes=master-a,master-b,master-c
--------------------------------------------------

It is technically sufficient to set this on a single master-eligible node in
the cluster, and only to mention that single node in the setting, but this
provides no fault tolerance before the cluster has fully formed. It
is therefore better to bootstrap using at least three master-eligible nodes.
In any case, when specifying the list of initial master nodes, **it is vitally
important** to configure each node with exactly the same list of nodes, to
prevent two independent clusters from forming. Typically you will set this on
the nodes that are mentioned in the list of initial master nodes.

NOTE: In alpha releases, all listed master-eligible nodes are required to be
  discovered before bootstrapping can take place. This requirement will be
  relaxed in production-ready releases.

> WARNING: You must put exactly the same set of initial master nodes in each

Reviewer: This seems like it's restating the "vitally important" info above, so I'd suggest removing the redundancy from one place or the other.

  configuration file (or leave the configuration empty) in order to be sure
  that only a single cluster forms during bootstrapping and therefore to
  avoid the risk of data loss.

[float]
==== Choosing a cluster name

> The `cluster.name` allows you to create multiple clusters which are separated

Reviewer: ```suggestion
The <<cluster.name,`cluster.name`>> setting enables you to create multiple clusters which are separated
```

from each other. Nodes verify that they agree on their cluster name when they

> first connect to each other, and if two nodes have different cluster names then

Reviewer: ```suggestion
first connect to each other. If two nodes have different cluster names, 
```

> they will not communicate meaningfully and will not belong to the same cluster.

Reviewer: Not sure what "communicate meaningfully" means. Can this be omitted?  If not, what's it implying?
